File: app/src/main/java/com/aaronbrecher/neverlate/ui/fragments/AppSnoozedFragment.java
package com.aaronbrecher.neverlate.ui.fragments;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.aaronbrecher.neverlate.R;
import com.aaronbrecher.neverlate.interfaces.NavigationControl;
import com.aaronbrecher.neverlate.ui.activities.MainActivity;

public class AppSnoozedFragment extends Fragment {
    private Button mButton;
    private NavigationControl mNavController;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        try {
            mNavController = (NavigationControl) getActivity();
        } catch (ClassCastException e) {
            throw new ClassCastException(context.toString() +
                    " must implement the ListItemClickListener interface");
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_app_snoozed, container, false);
        mButton = rootView.findViewById(R.id.app_snoozed_cancel_button);
        return rootView;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        mButton.setOnClickListener(v -> mNavController.navigateToDestination(R.id.snoozeFragment));
    }
}

File: app/src/main/java/com/aaronbrecher/neverlate/ui/fragments/AppSnoozedFragment.kt
package com.aaronbrecher.neverlate.ui.fragments

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button

import com.aaronbrecher.neverlate.R
import com.aaronbrecher.neverlate.interfaces.NavigationControl

import androidx.annotation.NonNull
import androidx.annotation.Nullable
import androidx.fragment.app.Fragment

class AppSnoozedFragment : Fragment() {
    private lateinit var mButton: Button
    private lateinit var mNavController: NavigationControl



    override fun onAttach(context: Context) {
        super.onAttach(context)
        try {
            mNavController = activity as NavigationControl
        } catch (e: ClassCastException) {
            throw ClassCastException(context.toString() + " must implement the ListItemClickListener interface")
        }

    }



    @Nullable
    override fun onCreateView(@NonNull inflater: LayoutInflater, @Nullable container: ViewGroup?, @Nullable savedInstanceState: Bundle?): View {
        val rootView = inflater.inflate(R.layout.fragment_app_snoozed, container, false)
        activity?.setTitle(R.string.app_snoozed_title)
        mButton = rootView.findViewById(R.id.app_snoozed_cancel_button)
        return rootView
    }

    override fun onViewCreated(@NonNull view: View, @Nullable savedInstanceState: Bundle?) {
        mButton.setOnClickListener { mNavController.navigateToDestination(R.id.snoozeFragment) }
    }
}
